packages/graphql/tests/integration/issues/4170.int.test.ts

```diff
@@ -0,0 +1,191 @@
+ */
+
+import type { Driver } from "neo4j-driver";
+import Neo4j from "../neo4j";
+import { Neo4jGraphQL } from "../../../src/classes";
+import gql from "graphql-tag";
+import { graphql } from "graphql";
+import { UniqueType } from "../../utils/graphql-types";
+
+describe("https://github.com/neo4j/graphql/issues/4170", () => {
+    let driver: Driver;
+    let neo4j: Neo4j;
+
+    const User = new UniqueType("User");
+    const Tenant = new UniqueType("Tenant");
+    const Settings = new UniqueType("Settings");
+    const OpeningDay = new UniqueType("OpeningDay");
+    const OpeningHoursInterval = new UniqueType("OpeningHoursInterval");
+
+    const typeDefs = gql`
+        type JWT @jwt {
+            id: String
+            roles: [String]
+        }
+        type ${User.name} @authorization(validate: [{ where: { node: { userId: "$jwt.id" } }, operations: [READ] }]) {
+            userId: String! @unique
+            adminAccess: [${Tenant.name}!]! @relationship(type: "ADMIN_IN", direction: OUT)
+        }
+
+        type ${Tenant.name} @authorization(validate: [{ where: { node: { admins: { userId: "$jwt.id" } } } }]) {
+            id: ID! @id
+            settings: ${Settings.name}! @relationship(type: "HAS_SETTINGS", direction: OUT)
+            admins: [${User.name}!]! @relationship(type: "ADMIN_IN", direction: IN)
+        }
+
+        type ${Settings.name} @authorization(validate: [{ where: { node: { tenant: { admins: { userId: "$jwt.id" } } } } }]) {
+            id: ID! @id
+            tenant: ${Tenant.name}! @relationship(type: "HAS_SETTINGS", direction: IN)
+            openingDays: [${OpeningDay.name}!]! @relationship(type: "VALID_OPENING_DAYS", direction: OUT)
+            name: String
+        }
+
+        type ${OpeningDay.name}
+            @authorization(
+                validate: [{ where: { node: { settings: { tenant: { admins: { userId: "$jwt.id" } } } } } }]
+            ) {
+            id: ID! @id
+            settings: ${Settings.name} @relationship(type: "VALID_GARAGES", direction: IN)
+            open: [${OpeningHoursInterval.name}!]! @relationship(type: "HAS_OPEN_INTERVALS", direction: OUT)
+        }
+
+        type ${OpeningHoursInterval.name}
+            @authorization(
+                validate: [
+                    { where: { node: { openingDay: { settings: { tenant: { admins: { userId: "$jwt.id" } } } } } } }
+                ]
+            ) {
+            name: String
+            openingDay: ${OpeningDay.name}! @relationship(type: "HAS_OPEN_INTERVALS", direction: IN)
+            updatedBy: String @populatedBy(callback: "getUserIDFromContext", operations: [CREATE, UPDATE])
+        }
+    `;
+
+    const ADD_TENANT = `
+        mutation addTenant($input: [${Tenant.name}CreateInput!]!) {
+            ${Tenant.operations.create}(input: $input) {
+                ${Tenant.plural} {
+                    id
+                    admins {
+                        userId
+                    }
```

Comment:
If you also  query
```
                                settings {
                                    openingDays {
                                        open {
                                            name
                                        }
                                    }
                                }
```
here, you will get error `Neo4jError: Variable `create_var9` not defined (line 108, column 80 (offset: 6513))`
Just noticed  this in my project, when I tried applying this `idx + 1` fix, should maybe be a different issue if the cause is very different.

Reply:
hi @afreakk, yes it is definitely a different issue but I will look into that as well. Thanks for flagging it!